platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -1087,11 +1109,18 @@ namespace platoon_strategic_ihp
                 //  -ensure plan type corresponds to join_index
 
                 // complete the request
-
-                int join_index = -1;
                 // UCLA: this is a newly added plan type 
+                std::string newLeaderPlatoonId = pm_.currentPlatoonID; // set self's (front join veh) platoon ID as the new platoon ID
+                ROS_DEBUG_STREAM("Host vehicle platoon id = " << newLeaderPlatoonId << "; join the rear platoon as the new leader (frontal join).");
+
+                // Note: Request conposed outside of if conditions
+                // UCLA: Desired joining index for cut-in join, indicate the index of gap-leading vehicle. -1 indicate cut-in from front.
+                // Note: remove join_index to info param.
                 request.plan_type.type = cav_msgs::PlanType::CUT_IN_FROM_SIDE; 
 
+                // note: At this step all cut-in types all start with this request, so the join_index at this point is set to default, -2.
+                // REVISIOn note: At this step a join index is really  not needed until later. The later index were changed to name "target_index". So it is ok to delete cut-in here.
```

Comment:
Seems like this should be a TODO  that needs to be resolved before we are ready to test cut-ins.

Reply:
Indeed.